### PR TITLE
feat: Add ability to use secure WebSocket to listen Logcat messages

### LINKED ISSUE #2181: startLogcatBroadcast does not respect https
## Description

Bug.
The Appium is accessible over **HTTPS**.
```
curl https://<user>:<pass>@<host>:8443/status
{"value":{"ready":true,"message":"The server is ready to accept new connections","build":{"version":"2.1.3"}}}
```
The tests are triggered with java property -Dappium.hub=**https**://user:pass@host:8443.

When we call `startLogcatBroadcast`:
`androidDriver.startLogcatBroadcast(androidDriver.getRemoteAddress().getHost(), androidDriver.getRemoteAddress().getPort());
`
We get error:
```
org.openqa.selenium.remote.http.ConnectionFailedException: JdkWebSocket initial request execution error
Build info: version: '4.21.0', revision: '79ed462ef4'
System info: os.name: 'Mac OS X', os.arch: 'x86_64', os.version: '14.5', java.version: '17.0.7'
Driver info: driver.version: unknown
Caused by: java.net.http.WebSocketHandshakeException
Caused by: jdk.internal.net.http.websocket.CheckFailedException: Unexpected HTTP response status code 404
	... 11 more
java.base/java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:396)
java.base/java.util.concurrent.CompletableFuture.get(CompletableFuture.java:2096)
org.openqa.selenium.remote.http.jdk.JdkHttpClient.openSocket(JdkHttpClient.java:240)
io.appium.java_client.ws.StringWebSocketClient.connect(StringWebSocketClient.java:74)
io.appium.java_client.android.ListensToLogcatMessages.startLogcatBroadcast(ListensToLogcatMessages.java:67)
```

URI: **http**://host:8443/ws/session/b515a27e-ff21-4e43-b8f2-b336eed55665/appium/device/logcat

[Here](https://github.com/appium/java-client/blob/e62b3951cb48bea076d104a3a3d0b6588e618587/src/main/java/io/appium/java_client/android/ListensToLogcatMessages.java#L62C13-L62C24) the endpointUri is set to `ws://` regardless  `-Dappium.hub=https://`. The `wss://` should be used.

## Environment

* Java client build version: java-client-9.2.3.jar
* Appium server version: 2.1.3
* OS version used to run Appium: Ubuntu 18.04.3 LTS
* Node.js version: v16.20.2
* Mobile platform/version under test: Android
* Real device: moto-e40 (Android 11)

### PR DESCRIPTION
**Pull request by valfirst**
## Change list

Closes #2181.
 
## Types of changes

What types of changes are you proposing/introducing to Java client?
_Put an `x` in the boxes that apply_

- [ ] No changes in production code.
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: src/main/java/io/appium/java_client/android/ListensToLogcatMessages.java [hunk 1]
```diff
@@ -49,21 +49,42 @@ default void startLogcatBroadcast(String host) {
...
-        final URI endpointUri;
-        try {
-            endpointUri = new URI(String.format("ws://%s:%s/ws/session/%s/appium/device/logcat",
-                    host, port, ((RemoteWebDriver) this).getSessionId()));
-        } catch (URISyntaxException e) {
-            throw new IllegalArgumentException(e);
-        }
+        String scheme = useSecureConnection ? "wss" : "ws";
+        SessionId sessionId = ((RemoteWebDriver) this).getSessionId();
+        URI endpointUri = URI.create(String.format("%s://%s:%s/ws/session/%s/appium/device/logcat", scheme, host,
```
[the remark this answers is not in the document]
Comment (valfirst) [reply src/main/java/io/appium/java_client/android/ListensToLogcatMessages.java:86]: yes, I think, we can determine the scheme if server URL is available, but what will be a reliable way to get it? I can invent something like:
```java
URL serverUrl = ((HttpCommandExecutor)((RemoteWebDriver) this).getCommandExecutor()).getAddressOfRemoteServer()
```
Also having this URL, is it needed to keep methods like:
```
startLogcatBroadcast(String host)
startLogcatBroadcast(String host, int port)
```
?

**Comment by janmekota** [comment]
Probably [here](https://github.com/appium/java-client/blob/e62b3951cb48bea076d104a3a3d0b6588e618587/src/main/java/io/appium/java_client/ios/ListensToSyslogMessages.java#L63) will be a similar issue.

File: src/main/java/io/appium/java_client/android/ListensToLogcatMessages.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (mykola-mokhnach) [reply src/main/java/io/appium/java_client/android/ListensToLogcatMessages.java:86]: Sorry, the PR got lost in the stream. @valfirst feel free to ping me in slack if you need a review

we just check if the URLs protocol is https. that is enough. No other changes or overloads would be needed with that

File: src/main/java/io/appium/java_client/android/ListensToLogcatMessages.java [hunk 2]
```diff
@@ -56,15 +58,13 @@ default void startLogcatBroadcast(String host) {
      * @param port the port of the host where Appium server is running
      */
     default void startLogcatBroadcast(String host, int port) {
+        RemoteWebDriver remoteWebDriver = (RemoteWebDriver) this;
```
Comment (mykola-mokhnach) [inline src/main/java/io/appium/java_client/android/ListensToLogcatMessages.java:61]: you could use `var` instead of explicit typedefs

**Review by mykola-mokhnach — APPROVED** [review]

**Comment by jlipps** [comment]
Hi @valfirst , congrats, the Appium project wants to compensate you for this (and perhaps other) contribution(s) this month! Please reply to this comment mentioning me and sharing your [OpenCollective](https://opencollective.com) account name, so that we can initiate payment! Or let me know if you decline to receive compensation via OpenCollective. Thank you!

**Comment by valfirst** [comment]
Hi @jlipps, here is a link to my account: https://opencollective.com/valfirst

**Pull request merged**